packages/cli/src/executions/execution.service.ts

```diff
@@ -392,67 +390,81 @@ export class ExecutionService {
 		};
 	}
 
-	/**
-	 * Stop an active execution.
-	 */
-	async stop(executionId: string) {
-		const execution = await this.executionRepository.findOneBy({ id: executionId });
+	async findAllEnqueuedExecutions() {
+		return await this.executionRepository.findMultipleExecutions(
+			{
+				select: ['id', 'mode'],
+				where: { status: 'new' },
+				order: { id: 'ASC' },
+			},
+			{ includeData: true, unflattenData: true },
+		);
+	}
+
+	async stop(executionId: string): Promise<StopResult> {
+		const execution = await this.executionRepository.findSingleExecution(executionId, {
+			includeData: true,
+			unflattenData: true,
+		});
+
+		if (!execution) throw new MissingExecutionStopError(executionId);
 
-		if (!execution) throw new NotFoundError('Execution not found');
+		this.assertStoppable(execution);
+
+		const { mode, startedAt, stoppedAt, finished, status } =
+			config.getEnv('executions.mode') === 'regular'
+				? await this.stopInRegularMode(execution)
+				: await this.stopInScalingMode(execution);
+
+		return {
+			mode,
+			startedAt: new Date(startedAt),
+			stoppedAt: stoppedAt ? new Date(stoppedAt) : undefined,
+			finished,
+			status,
+		};
+	}
 
-		if (execution.status === 'new') {
-			this.concurrencyControl.remove({ mode: execution.mode, executionId });
-			await this.executionRepository.cancel(executionId);
+	private assertStoppable(execution: IExecutionResponse) {
+		const STOPPABLE_STATUSES: ExecutionStatus[] = ['new', 'unknown', 'waiting', 'running'];
 
-			return;
+		if (!STOPPABLE_STATUSES.includes(execution.status)) {
+			throw new WorkflowOperationError(
+				`Only running or waiting executions can be stopped and ${execution.id} is currently ${execution.status}`,
+			);
 		}
+	}
 
-		const stopResult = await this.activeExecutions.stopExecution(execution.id);
+	private async stopInRegularMode(execution: IExecutionResponse) {
+		if (this.concurrencyControl.has(execution.id)) {
+			this.concurrencyControl.remove({ mode: execution.mode, executionId: execution.id });
+			return await this.executionRepository.stopBeforeRun(execution);
+		}
 
-		if (stopResult) return this.toExecutionStopResult(execution);
+		if (this.activeExecutions.has(execution.id)) {
+			await this.activeExecutions.stopExecution(execution.id);
+		}
 
-		if (this.isRegularMode) {
-			return await this.waitTracker.stopExecution(execution.id);
+		if (this.waitTracker.has(execution.id)) {
+			await this.waitTracker.stopExecution(execution.id);
 		}
 
-		// queue mode
+		return await this.executionRepository.stopDuringRun(execution);
+	}
 
-		try {
-			return await this.waitTracker.stopExecution(execution.id);
-		} catch {
-			// @TODO: Why are we swallowing this error in queue mode?
+	private async stopInScalingMode(execution: IExecutionResponse) {
```

Comment:
What if it's a manual execution? In this case it would be viewable in `activeExecutions` and can be cancelled from there.

Reply:
Please see [dcb93be](https://github.com/n8n-io/n8n/pull/9841/commits/dcb93be2bc7e0ab1b9b2c7cc9c6a78f822e136f5)